atlasdb-client/src/main/java/com/palantir/atlasdb/transaction/impl/ConflictDetectionManager.java

```diff
@@ -15,56 +15,133 @@
  */
 package com.palantir.atlasdb.transaction.impl;
 
-import java.util.Collections;
-import java.util.HashMap;
 import java.util.Map;
-import java.util.function.Supplier;
 
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
 import com.google.common.collect.Maps;
+import com.palantir.atlasdb.keyvalue.api.KeyValueService;
 import com.palantir.atlasdb.keyvalue.api.TableReference;
+import com.palantir.atlasdb.table.description.TableMetadata;
 import com.palantir.atlasdb.transaction.api.ConflictHandler;
 
 public class ConflictDetectionManager {
-    private final Supplier<Map<TableReference, ConflictHandler>> supplier;
-    private final Map<TableReference, ConflictHandler> overrides;
-    private final RecomputingSupplier<Map<TableReference, ConflictHandler>> cachedResult;
+    private static final Logger log = LoggerFactory.getLogger(ConflictDetectionManager.class);
+    private final LoadingCache<TableReference, ConflictHandler> cache;
```

Comment:
Actually, nvm. Now I see that this class works on top of the KeyValueService API